src/opensearch_sdk_py/settings/settings.py

```diff
@@ -0,0 +1,42 @@
+#
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+#
+
+
+from typing import Any, Dict, Optional, Union
+
+from opensearch_sdk_py.transport.stream_input import StreamInput
+from opensearch_sdk_py.transport.stream_output import StreamOutput
+
+
+class Settings:
+    def __init__(
+        self,
+        settings: dict[str, Union[str, Dict]] = {},
+        # TODO: Secure Settings
+    ) -> None:
+        self.settings = settings
```

Comment:
When initialized with a `Dict` I think you don't get a copy, does it behave like so?

```
initial = { 'x': 'y' }
settings = Settings(initial)
settings.read_from(...)
print(initial['Z']) # yields results
```

If it does, we should copy the dictionary or remove initializing with an external argument altogether.

Reply:
The initial static method implementation didn't have that problem. :)

We could always just empty the map on read_from.